src/library/export/trackexport.cpp

```diff
@@ -0,0 +1,129 @@
+#include "library/export/trackexport.h"
+
+#include <QFileInfo>
+#include <QMessageBox>
+#include <QDebug>
+
+void TrackExport::run() {
+    int i = 0;
+    for (const auto& track : m_tracks) {
+        QString path = track->getLocation();
+        QFileInfo fileinfo(path);
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
+        exportTrack(track);
+        if (m_bStop) {
+            emit(canceled());
+            return;
+        }
+        ++i;
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
```

Comment:
Why the double emit? We'll loop around and emit the next one "shortly" after though I guess QFileInfo does file IO so if there are some kind of stall it would delay updating to the i+1'th progress update?